File: AtlasWeb/Core/wwwroot/app/NebulaClient/StorageManager/DefinitionManagement/TTObjectDefBase.ts
import { Guid } from "../../Mscorlib/Guid";

export class TTObjectDefBase {

    private _objectDefID: Guid;
    public get ID(): Guid {
        return this._objectDefID;
    }

    protected _name: string;
    public get Name(): string {
        return this._name;
    }
    public set Name(value: string) {
        this._name = value;
    }

    public IsOfType(objectDefName: string): boolean;
    public IsOfType(objectDefID: Guid ): boolean;
    public IsOfType(objectDef: TTObjectDefBase ): boolean;
    public IsOfType(firstParam: string | Guid | TTObjectDefBase ): boolean {
        return null;
    }
}

File: AtlasModel/Enterprise/TTObjectDefBase.cs
using AtlasModel.Enterprise.Enums;
using System;

namespace AtlasModel.Enterprise
{
    public partial class TTObjectDefBase
    {
        public string ObjectDefId { get; set; }
        public string Name { get; set; }
        public Guid? BaseObjectDefId { get; set; }
        public bool IsAbstract { get; set; }
        public TTObjectSubTypeEnum SubType { get; set; }
        public string Methods { get; set; }
        public Guid ModuleDefId { get; set; }
        public string CheckOutId { get; set; }
        public CheckOutStatusEnum CheckOutStatus { get; set; }
        public string DisplayText { get; set; }
        public string HelpNamespace { get; set; }
        public string Description { get; set; }
        public string ObjectDefName { get; set; }
    }
}